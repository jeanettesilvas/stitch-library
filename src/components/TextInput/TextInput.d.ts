/// <reference types="react" />
import './TextInput.scss';
interface TextInputProps {
    className?: string;
    placeholder?: string;
    type?: 'text' | 'password' | 'email';
    value?: string;
}
export declare const TextInput: (props: TextInputProps) => JSX.Element;
export {};
