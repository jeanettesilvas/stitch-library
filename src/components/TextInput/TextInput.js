import React from 'react'
import PropTypes from 'prop-types'

import './TextInput.scss'

export const TextInput = props => {
  return (
    <input
      className={`text-input ${props.className}`}
      placeholder={props.placeholder}
      type={props.type}
      value={props.value}
    />
  )
}

TextInput.propTypes = {
  className: PropTypes.string,
  placeholder: PropTypes.string,
  type: PropTypes.oneOf(['password', 'text', 'email']),
  value: PropTypes.string,
}

TextInput.defaultProps = {
  className: '',
  type: 'text',
}
