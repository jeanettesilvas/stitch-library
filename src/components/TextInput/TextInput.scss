@import '../../style/global.scss';

.text-input {
  border: 1px solid $gray-c;
  border-radius: 4px;
  box-sizing: border-box;
  color: $gray-5;
  font-family: $base-font;
  font-size: 16px;
  padding: $spacing-1x $spacing-2x;
  transition: all 0.2s;

  &:focus {
    border-color: #000;
    box-shadow: 0 0 5px 0 #ccc;
    outline: none;
  }
}
