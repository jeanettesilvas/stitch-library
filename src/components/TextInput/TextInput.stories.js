import React from 'react'
import { storiesOf } from '@storybook/react'
import { TextInput } from './TextInput'

storiesOf('TextInput', module)
  .addDecorator(story => (
    <div style={{ marginTop: '50px', textAlign: 'center' }}>{story()}</div>
  ))
  .add('default', () => <TextInput />)
  .add('with placeholder', () => <TextInput placeholder="write something" />)
  .add('password', () => <TextInput type="password" value="password" />)
  .add('email', () => <TextInput type="email" placeholder="email@xyz.com" />)
