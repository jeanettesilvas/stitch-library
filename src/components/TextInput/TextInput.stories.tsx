import React from 'react'
import { Story, Meta } from '@storybook/react'

import { TextInput, TextInputProps } from './TextInput'

export default {
  title: 'Components/TextInput',
  component: TextInput,
  argTypes: {
    className: { control: '' },
    placeholder: { control: '' },
    type: { control: 'text' },
    value: { control: '' },
  },
} as Meta

const Template: Story<TextInputProps> = (args) => <TextInput {...args} />

export const Text = Template.bind({})
Text.args = {
  placeholder: 'Enter text',
  type: 'text',
}

export const Password = Template.bind({})
Password.args = {
  placeholder: 'Enter password',
  type: 'password',
}

export const Email = Template.bind({})
Email.args = {
  placeholder: 'Enter email',
  type: 'email',
}
