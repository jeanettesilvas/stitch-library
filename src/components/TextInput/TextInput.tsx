import './TextInput.scss'

export interface TextInputProps {
  className?: string
  placeholder?: string
  type?: 'text' | 'password' | 'email'
  value?: string
}

export const TextInput = (props: TextInputProps) => {
  let { className, placeholder, type, value } = props

  return (
    <input
      className={`text-input ${className}`}
      placeholder={placeholder}
      type={type}
      value={value}
    />
  )
}
