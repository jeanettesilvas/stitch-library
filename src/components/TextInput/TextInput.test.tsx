import renderer from 'react-test-renderer'

import { TextInput } from './TextInput'

describe('Text', () => {
  it('renders', () => {
    const tree = renderer.create(<TextInput onChange={() => {}} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
