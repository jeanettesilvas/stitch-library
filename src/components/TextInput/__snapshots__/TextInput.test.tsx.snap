// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders 1`] = `
<div
  className="text-input text-input--default"
>
  <input
    className="text-input__field text-input__field--default"
    onChange={[Function]}
    value=""
  />
  <span
    className="text-input__message text-input__message--default"
  >
    
  </span>
</div>
`;
